api/src/main/java/org/apache/gravitino/model/Model.java

```diff
@@ -0,0 +1,171 @@
+   *
+   * @return the properties of the model object.
+   */
+  default Map<String, String> properties() {
+    return Collections.emptyMap();
+  }
+
+  /**
+   * The versions of the model object. The model object can have multiple versions. Each version
+   * contains the detailed information of the model artifact, like the URL, the training properties,
+   * etc.
+   *
+   * @return The versions of the model object.
+   */
+  ModelVersion[] versions();
+
+  /**
+   * Get the model version by the version number. If the version does not exist, throw an exception.
+   *
+   * @param version The version number of the model.
+   * @return The model version object.
+   * @throws NoSuchModelVersionException if the version does not exist.
+   */
+  ModelVersion version(int version) throws NoSuchModelVersionException;
+
+  /**
+   * Get the model version by the version alias. If the version does not exist, throw an exception.
+   *
+   * @param alias The version alias of the model.
+   * @return The model version object.
+   * @throws NoSuchModelVersionException if the version does not exist.
+   */
+  ModelVersion versionAlias(String alias) throws NoSuchModelVersionException;
+
+  /**
+   * Check if the model version exists by the version number.
+   *
+   * @param version The version number of the model.
+   * @return True if the version exists, false otherwise.
+   */
+  default boolean versionExists(int version) {
+    try {
+      version(version);
+      return true;
+    } catch (NoSuchModelVersionException e) {
+      return false;
+    }
+  }
+
+  /**
+   * Check if the model version exists by the version alias.
+   *
+   * @param alias The version alias of the model.
+   * @return True if the version exists, false otherwise.
+   */
+  default boolean versionExists(String alias) {
+    try {
+      versionAlias(alias);
+      return true;
+    } catch (NoSuchModelVersionException e) {
+      return false;
+    }
+  }
+
+  /**
+   * Link a new version model to the registered model object. The new version model will be added to
+   * the model object. If the model object does not exist, it will throw an exception.
+   *
+   * @param aliases The aliases of the model version. The aliases should be unique in this model,
+   *     otherwise the {@link ModelVersionAliasesAlreadyExistsException} will be thrown. The aliases
+   *     are optional and can be empty.
+   * @param comment The comment of the model version. The comment is optional and can be null.
+   * @param url The model artifact URL.
+   * @param properties The properties of the model version. The properties are optional and can be
+   *     null or empty.
+   * @return The created model version object.
+   * @throws NoSuchModelException If the model does not exist.
+   * @throws ModelVersionAliasesAlreadyExistsException If the aliases already exist in the model.
+   */
+  ModelVersion link(String[] aliases, String comment, String url, Map<String, String> properties)
```

Comment:
Do we have the interface `unlink` to change the version alias?

Reply:
For the change/alter/update support, I will do this in the next version. In this version, we only support register/link, get, list and delete operations, so I didn't add that interface.